assets/javascripts/kitten/components/uploaders/uploader.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react'
+import classNames from 'classnames'
+
+export class Uploader extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      errors: {
+        extensionsError: false,
+        sizeError: false,
+        hasError: false,
+      },
+    }
+
+    this.handleChange = this.handleChange.bind(this)
+  }
+
+  checkExtensions(file) {
+    if (!this.props.acceptedFiles) return
+
+    const extensions = this.props.acceptedFiles.split(',')
+    const regex = new RegExp(`(${ extensions.join('|') })`)
+
+    this.setState({
+      errors: {
+        ...this.state.errors,
+        extensionsError: file.type.search(regex) < 1,
+      }
+    })
+  }
+
+  checkSize(file) {
+    this.setState({
+      errors: {
+        ...this.state.errors,
+        sizeError: this.props.maxSize && file.size > this.props.maxSize,
+      }
+    })
+  }
+
+  updateHasError() {
+    const errors = this.state.errors
+
+    this.setState({
+      errors: {
+        ...errors,
+        hasError: errors.extensionsError || errors.sizeError,
```

Comment:
J'ai l'impression que cette information n'a pas besoin d'être dans le state car ça ne contient qu'une duplication de l'état.

Une fonction `hasError() { return errors.extensionsError || errors.sizeError }` pourrait remplacer `this.state.hasError` par `this.hasError()`.